pcb/kicad_mod.py

```diff
@@ -701,143 +635,350 @@ def filterPads(self, pad_type):
         for pad in self.pads:
             if pad['type'] == pad_type:
                 pads.append(pad)
+                
+        pads = sorted(pads, key = lambda p : str(p['number']))
 
         return pads
+        
+    # Get the middle position between pads
+    def padMiddlePosition(self, pads=None):
```

Comment:
This function should be reformulated to:

```python
def padMiddlePosition(self, pads=None):
    bb = self.padsBounds(pads)
    return bb.center
```

Otherwise this function calculates the center of mass of the center-positions of each pad. If then the pads are distributed unevenly between left and right (e.g. see https://github.com/KiCad/Pin_Headers.pretty/blob/master/Pin_Header_Straight_1x39_Pitch2.54mm_SMD_Pin1Left.kicad_mod ) the center pin is asked to be ofset a bit (0.05mm to the left or right from the (IMHO correct) center-line